lib/src/iterable_extensions.dart

```diff
@@ -589,6 +589,19 @@ extension IterableExtension<T> on Iterable<T> {
     }
     return true;
   }
+
+  /// The elements with [seperator] between each.
```

Comment:
This documentation should be more elaborate.
Something like:
```dart
/// The elements of [this] interleaved by [separator].
///
/// Inserts [separator] between any two elements of this iterable.
/// If this iterable is empty or has only one element, no separators are inserted,
/// otherwise a separator is inserted before each element of this iterable
/// other than the first.
```
(so we cover the the edge cases explicitly).